pkg/search/postgres/store.go

```diff
@@ -386,6 +389,39 @@ func (s *genericStore[T, PT]) DeleteMany(ctx context.Context, identifiers []stri
 	return nil
 }
 
+// PruneMany removes the objects associated to the specified IDs from the store outside a transaction.
+func (s *genericStore[T, PT]) PruneMany(ctx context.Context, identifiers []string) error {
+	defer s.setPostgresOperationDurationTime(time.Now(), ops.RemoveMany)
+
+	// Batch the deletes
+	localBatchSize := pruneBatchSize
+
+	for {
+		if len(identifiers) == 0 {
+			break
+		}
+
+		if len(identifiers) < localBatchSize {
+			localBatchSize = len(identifiers)
+		}
+
+		identifierBatch := identifiers[:localBatchSize]
+
+		q := search.NewQueryBuilder().AddDocIDs(identifierBatch...).ProtoQuery()
+
+		// If a pruning batch fails, just move on to the next batch as pruning is best effort and eventual.
+		if err := RunDeleteRequestForSchema(ctx, s.schema, q, s.db); err != nil {
+			log.Errorf("unable to prune the records: %v", err)
+			continue
+		}
+
+		// Move the slice forward to start the next batch
+		identifiers = identifiers[localBatchSize:]
+	}
```

Comment:
Would it be possible to extract the logic common to `DeleteMany` and `PruneMany` in an internal function ?
I think with parameters such as `initialBatchSize` and `continueOnError` most of the logic could be extracted.

Reply:
Good idea.  Take a look at what I did so see if it looks right.